aten/src/ATen/native/mps/operations/Distributions.mm

```diff
@@ -456,5 +456,70 @@ static void check_from_to_in_range(int64_t from, int64_t to_inc, ScalarType scal
   return random_mps_(self, 0, to, gen);
 }
 
+// Exponential distribution
+
+Tensor& exponential_mps_(Tensor& self, double lambda, c10::optional<Generator> gen) {
+
+  using namespace mps;
+
+  if (self.numel() == 0) {
+    return self;
+  }
+
+  TORCH_CHECK(lambda > 0, "exponential_mps_: lambda must be greater than zero")
+
+  struct CachedGraph : public MPSCachedGraph
+  {
+    CachedGraph(MPSGraph *graph) : MPSCachedGraph(graph) {}
+    MPSGraphTensor *outputTensor_ = nil;
+  };
+
+  MPSGraphCache* cache_ = MPSGraphCache::getInstance();
+
+  MPSStream* stream = getCurrentMPSStream();
+  uint64_t seed_ = c10::detail::getNonDeterministicRandom(true);
+
+  @autoreleasepool {
+    MPSShape* self_shape = getMPSShape(self);
+
+    MPSGraph* mpsGraph = make_mps_graph();
+          
+    // TODO: right now taking the default seed. Extend it to be extracted from the
+    // MPSGenerator
+    MPSGraphTensor* randomTensor = [mpsGraph randomUniformTensorWithShape:self_shape
```

Comment:
These are all the ops in PyTorch which don't have Graph caching. @abhudev , can we prioritize moving all of these ops to new MPS random uniform API.?